engine/dlib/src/dlib/job_thread.cpp

```diff
@@ -109,48 +110,109 @@ static inline uint32_t ToIndex(HJob job)
     return job & 0xFFFFFFFF;
 }
 
-// ***********************************************************************************
-// Jobs
-
-static void FreeJob(JobThreadContext* ctx, HJob hjob)
-{
-    uint32_t generation = ToGeneration(hjob);
-    uint32_t index = ToIndex(hjob);
-    JobItem& item = ctx->m_Items.Get(index);
-    assert(item.m_Generation == generation);
-    item.m_Generation = INVALID_INDEX;
-    item.m_Status = JOB_STATUS_FREE;
-
-    ctx->m_Items.Free(index, false);
-}
-
-static JobItem* ToItem(JobThreadContext* ctx, HJob hjob)
+static JobItem* GetJobItem(JobThreadContext* ctx, HJob hjob)
 {
     uint32_t generation = ToGeneration(hjob);
     uint32_t index      = ToIndex(hjob);
-    JobItem* item = &ctx->m_Items.Get(index);
-    if (item->m_Generation != generation)
-    {
+    JobItem* item = ctx->m_Items.GetPtr(index);
+    if (item == 0 || item->m_Generation != generation)
         return 0;
-    }
     return item;
 }
 
 static JobItem* CheckItem(JobThreadContext* ctx, HJob hjob)
 {
-    JobItem* item = ToItem(ctx, hjob);
+    JobItem* item = GetJobItem(ctx, hjob);
     assert(item != 0); // Generation differed!
     return item;
 }
 
-static JobItem* GetJobItem(JobThreadContext* ctx, HJob hjob)
+// ***********************************************************************************
+// Jobs
+
+static void RemoveChildFromParent(JobThreadContext* ctx, HJob hchild)
```

Comment:
Bug fix: Added this as the linked list otherwise would contain dangling "references", in the middle of the list, and the rest of the list couldn't be traversed.